application/Common/Shared/Utility/NativeMethods.cs

```diff
@@ -75,42 +74,73 @@ public static void StopMessageLoop()
+        /// <returns><see langword="true" /> if the key is pressed, <see langword="false" /> otherwise.</returns>
+        public static bool IsKeyPressed(VirtualKeyCode virtualKeyCode)
+        {
+            return Convert.ToBoolean(GetKeyState(virtualKeyCode) & (int) KeyMask.KEY_PRESSED);
+        }
+
+        #endregion
+
         #region SetHook helper
 
+        private static bool TryGetCurrentModuleHandle(out IntPtr handle)
+        {
+            using var currentProcess = Process.GetCurrentProcess();
+            using var currentModule = currentProcess.MainModule;
+            if (currentModule == null)
+            {
+                handle = IntPtr.Zero;
+                return false;
+            }
+
+            handle = GetModuleHandle(currentModule.ModuleName);
+            return true;
+        }
+
         /// <summary>
         ///     Sets a low-level keyboard hook.
         /// </summary>
         /// <param name="callback">The callback of the hook.</param>
         /// <param name="handle">The handle of the hook. Valid if the method returns <see langword="true" /></param>
         /// <returns><see langword="true" /> if the hook could successfully be set, <see langword="false" /> otherwise.</returns>
-        public static bool TrySetKeyboardHook(LowLevelKeyboardProc callback, [NotNullWhen(true)] out IntPtr handle)
+        public static bool TrySetKeyboardHook(LowLevelKeyboardProc callback, out IntPtr handle)
         {
-            using var currentProcess = Process.GetCurrentProcess();
-            using var currentModule = currentProcess.MainModule;
-
-            if (currentModule == null)
+            if (!TryGetCurrentModuleHandle(out var moduleHandle))
             {
                 handle = IntPtr.Zero;
                 return false;
             }
 
-            var moduleHandle = GetModuleHandle(currentModule.ModuleName);
+
             handle = SetWindowsHookEx(HookType.WH_KEYBOARD_LL, callback, moduleHandle, 0);
             return true;
         }
 
-        #endregion
-
-        #region Keyboard state helper
 
         /// <summary>
-        ///     Indicates whether a key identified by its virtual key code is pressed
+        ///     Sets a low-level mouse hook.
         /// </summary>
-        /// <param name="virtualKeyCode">The <see cref="VirtualKeyCode" /> of the key to check.</param>
-        /// <returns><see langword="true" /> if the key is pressed, <see langword="false" /> otherwise.</returns>
-        public static bool IsKeyPressed(VirtualKeyCode virtualKeyCode)
+        /// <param name="callback">The callback of the hook.</param>
+        /// <param name="handle">The handle of the hook. Valid if the method returns <see langword="true" /></param>
+        /// <returns><see langword="true" /> if the hook could successfully be set, <see langword="false" /> otherwise.</returns>
+        public static bool TrySetMouseHook(LowLevelMouseProc callback, out IntPtr handle)
```

Comment:
This is currently unused, if you don't end up setting the low-level hook again (see below), you could remove this